### PR TITLE
Caution about impersonation not compatible with pre authenticated

### PR DESCRIPTION
**Pull request by pasdeloup**
Caution added to close Symfony issue #2172 according to @fabpot comment in the Symfony #19059 PR.
I also improved the similar caution about REMOTE_USER to make it global.

### CODE REVIEW

File: cookbook/security/impersonating_user.rst [hunk 1]
```diff
@@ -6,8 +6,17 @@ How to Impersonate a User
...
-to understand a bug a user sees that you can't reproduce). This can be easily
-done by activating the ``switch_user`` firewall listener:
+to understand a bug a user sees that you can't reproduce).
+
+.. caution::
+
+    User impersonation is not compatible with
+    :doc:`pre Authenticated firewalls</cookbook/security/pre_authenticated>`. The
+    reason is that impersonation requires the authentication state to be maintained
+    server-side but pre Authenticated information (``SSL_CLIENT_S_DN_Email``,
```
Comment (javiereguiluz) [inline cookbook/security/impersonating_user.rst]: `pre Authenticated information` -> `preauthenticated information` or `pre-authenticated information`

File: cookbook/security/pre_authenticated.rst [hunk 2]
```diff
@@ -11,6 +11,14 @@ box, Symfony supports most authentication mechanisms.
 These requests are called *pre authenticated* requests because the user is already
 authenticated when reaching your application.
 
+.. caution::
+
+    :doc:`User impersonation </cookbook/security/impersonating_user>` is not
+    compatible with pre Authenticated firewalls. The reason is that
```
Comment (javiereguiluz) [inline cookbook/security/pre_authenticated.rst]: `pre Authenticated firewalls ` -> `preauthenticated firewalls` or `pre-authenticated firewalls`

Comment (pasdeloup) [inline cookbook/security/pre_authenticated.rst]: It's what I would write also, but the title of the cookbook is "Using pre Authenticated Security Firewalls" so I wanted to be consistent.
https://symfony.com/doc/2.7/cookbook/security/pre_authenticated.html

Should it be changed everywhere?

Comment (javiereguiluz) [inline cookbook/security/pre_authenticated.rst]: @pasdeloup the reason is that we use a strange format for titles, but for regular text we don't mix upper and lower case.

Comment (pasdeloup) [inline cookbook/security/pre_authenticated.rst]: Ok fixed. I missed that rule about section titles
However, shouldn't the title be either

> Using pre-Authenticated Security Firewalls

or

> Using Preauthenticated Security Firewalls

Comment (javiereguiluz) [inline cookbook/security/pre_authenticated.rst]: @pasdeloup it's "complicated" to say if you are right or wrong. First, the title casing is arbitrary and some words are uppercased and others aren't. Moreover, the hyphen after prefixes is also arbitrary in some cases, so I don't know if it's written "preauth..." or "pre-auth..."

Comment (pasdeloup) [inline cookbook/security/pre_authenticated.rst]: Ok, let's keep like that for now ;)

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed

**Comment by xabbuh** [comment]
👍

**Comment by WouterJ** [comment]
Thank you @pasdeloup! This is a very nice caution, explaining just enough to know why this aren't supported without being too verbose.

I've made some very minor rewordings in dd3f08f after merging this PR. If you don't agree, feel free to comment. Thanks again!

**Comment by pasdeloup** [comment]
Thanks, it's better like that.

**Pull request merged**